runtime/callback.c

```diff
@@ -184,6 +184,12 @@ CAMLexport value caml_callback2_exn(value closure, value arg1, value arg2)
     value res;
 
     cont = save_and_clear_stack_parent(domain_state);
+
+    /* save_and_clear_stack_parent may trigger the GC and move values around.
+       so they are reloaded. */
+    args[0] = arg1;
+    args[1] = arg2;
```

Comment:
Unless I misunderstand, `args` is in fact not needed up to that point. Could we define it here (and also in the other branch then), instead of defining it above and then reloading?

Reply:
Yes it makes sense. Not using an array and separating parameters also makes sense, so I will implement the two options.